bin/test-merges.js

```diff
@@ -0,0 +1,296 @@
+const WORKING_DIR = process.env.AS_PINE_TEST_DIR;
+
+const SimpleGit = require("simple-git");
+const TESTING_GIT_AS = path.join(WORKING_DIR, AS_REPO_NAME);
+const git = new SimpleGit(TESTING_GIT_AS);
+
+const PREPARE_MOCHITESTS_DEV = path.join(TESTING_GIT_AS, "bin", "prepare-mochitests-dev");
+const TESTING_LOCAL_MC = path.join(WORKING_DIR, "mozilla-central");
+
+/**
+ * Find all PRs merged since ${OLDEST_PR_DATE} that don't have
+ * ${ALREADY_PUSHED_LABEL}
+ *
+ * @return {Promise} Promise that resolves with the search results or rejects
+ */
+function findNewlyMergedPRs() {
+  const searchTerms = [
+    // closed PRs in our repo
+    `repo:${AS_REPO}`, "type:pr", "state:closed", "is:merged",
+
+    // don't try and mochitest old closed stuff, we don't want to kick off a
+    // zillion test jobs
+    `merged:>=${OLDEST_PR_DATE}`,
+
+    // only look at merges to master
+    "base:master",
+
+    // if it's already been pushed to pine, don't do it again
+    `-label:${ALREADY_PUSHED_LABEL}`
+  ];
+
+  console.log(`Searching ${AS_REPO} for newly merged PRs`);
+  return github.search.issues({q: searchTerms.join("+")});
+}
+
+/**
+ * Return the commitId when the given PR was merged.  This is the one
+ * we will want to export and test.
+ *
+ * @param  {String} prNumber  The number of the PR to export.
+ * @return {String}           The commitId associated with the merge of this PR.
+ */
+function getPRMergeCommitId(prNumber) {
+  return github.issues.getEvents({
+    owner: AS_REPO_OWNER,
+    repo: AS_REPO_NAME,
+    issue_number: prNumber
+  }).then(data => {
+    if (data.incomplete_results) {
+      // XXX should handle this case theoretically, but since we'll be running
+      // regularly from cron, it seems unlikely that we'll even hit 30 new
+      // merges (default GitHub page size) in a single run.
+      throw new Error("data.incomplete_results is true, aborting");
+    }
+
+    let mergeEvents = data.filter(item => item.event === "merged");
+    if (mergeEvents.length > 1) {
+      throw new Error("more than one merge event, aborting");
+    }
+    let [mergeEvent] = mergeEvents;
+
+    if (!mergeEvent.commit_id) {
+      throw new Error("merge event has no commit id attached, aborted");
+    }
+    return mergeEvent.commit_id;
+  }).catch(reason => console.log(reason));
+}
+
+/**
+ * Checks out the given commit into ${TESTING_GIT_AS}
+ *
+ * @param  {String} commitId
+ * @return {Promise<String|?>}    resolves with commitId on checkout completion,
+ *                      rejects with error
+ */
+function checkoutGitCommit(commitId) {
+  return new Promise((resolve, reject) => {
+    console.log(`Fetching changes from github remote ${AS_REPO}...`);
+    // fetch any changes from the remote
+    git.fetch({}, (err, data) => {
+      if (err) {
+        reject(err);
+      }
+      console.log(`Starting github checkout of ${commitId}...`);
+      git.checkout(commitId, (err, data) => {
+        if (err) {
+          reject(err);
```

Comment:
Same as above. Does this need to be `return reject(err);` so that the Promise doesn't also try resolving on L134?